src/subplots.ts

```diff
@@ -2026,27 +2038,35 @@ export class Frame extends BasePlot {
     this.axes[1].transform(frameOrigin, yEnd);
   }
 
-  public setFeatures(data: any): [string, string] {
-    return [data.attribute_names[0], data.attribute_names[1]];
+  protected setFeatures(data: any): [string, string] {
+    let xFeature = data.attribute_names[0];
+    let yFeature = data.attribute_names[1];
+    if (!xFeature) { xFeature = "indices"; this.features.set("indices", Array.from(Array(this.nSamples).keys())) };
```

Comment:
`.keys()` is probably possibly unordered (I am unsure, though). Isn't that a problem in your case ?

Reply:
It is to build a range array when no abscissa is declared, like in plt.plot
https://stackoverflow.com/questions/3746725/how-to-create-an-array-containing-1-n

Reply:
Funnily enough, you linked the same thread I linked in a previous PR that made me have the exact same conclusion that the first comment : 

![image](https://github.com/Dessia-tech/plot_data/assets/42537234/8775d23f-d4c3-4616-909e-daa65c0ee6ed)

I guess it is ok though

Reply:
I agree

Reply:
For the discussed usecase, I think it is better with from since it is a one line instruction